charts/prometheus-pushgateway/templates/secret.yaml

```diff
@@ -8,3 +8,14 @@ metadata:
 data:
   web-config.yaml: {{ include "prometheus-pushgateway.webConfiguration" . | b64enc}}
 {{- end }}
+---
```

Comment:
The conditional `{{- if .Values.webConfiguration }}` has to be included in the second document, too.

The secret should only be created if `serviceMonitor.enabled` is set to _true_.

As a matter of fact, this change does not allow a user to set `serviceMonitor.endpoints.basicAuth` at all, e.g. for an existing secret (secrets get created by external mechanisms quite frequently). This field should also be supported independently of `webConfiguration`. I am thinking the secret should only be created if `serviceMonitor.endpoints.basicAuth` is not set by the user or even only if the user wants that secret created.